### PR TITLE
CLOUDP-70050: mongocli atlas security ldap save|create|update

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-70050

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
 ./bin/mongocli atlas security ldap save --hostname 2.tcp.ngrok.io --port 19657 --bindUsername "cn=admin,dc=example,dc=org" --binPassword admin
HOSTNAME PORT    AUTHENTICATION   AUTHORIZATION
2.tcp.ngrok.io   19657            true   true
```

```
~/workspace/mongoCLI git:(CLOUDP-70050) ✗ ./bin/mongocli atlas security ldap save --hostname 2.tcp.ngrok.io --port 19657 --bindUsername "cn=admin,dc=example,dc=org" --binPassword admin -o json
{
  "ldap": {
    "authenticationEnabled": true,
    "authorizationEnabled": true,
    "hostname": "2.tcp.ngrok.io",
    "port": 19657,
    "bindUsername": "cn=admin,dc=example,dc=org",
    "authzQueryTemplate": "{USER}?memberOf?base"
  }
}
```

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
waiting for https://github.com/mongodb/go-client-mongodb-atlas/pull/136

File: internal/cli/atlas/security/ldap/save.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				opts.initStore,
+				opts.InitOutput(cmd.OutOrStdout(), saveTemplate))
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.hostname, flag.Hostname, "", usage.LDAPHostname)
+	cmd.Flags().IntVar(&opts.port, flag.Port, 636, usage.LDAPPort)
```
Comment (colm-quinn) [inline internal/cli/atlas/security/ldap/save.go:93]: [q] Just to check, if you run the command with help, the default value will be shown?

File: internal/cli/atlas/security/ldap/save.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	cmd.Flags().StringVar(&opts.authzQueryTemplate, flag.AuthzQueryTemplate, "", usage.AuthzQueryTemplate)
+	cmd.Flags().BoolVar(&opts.authenticationEnabled, flag.AuthenticationEnabled, false, usage.AuthenticationEnabled)
+	cmd.Flags().BoolVar(&opts.authorizationEnabled, flag.AuthorizationEnabled, false, usage.AuthorizationEnabled)
+
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
+
+	_ = cmd.MarkFlagRequired(flag.Hostname)
+	_ = cmd.MarkFlagRequired(flag.BindUsername)
+	_ = cmd.MarkFlagRequired(flag.BindPassword)
```
Comment (colm-quinn) [inline internal/cli/atlas/security/ldap/save.go:106]: [q] The port is a required flag on the ticket - I see a default value is specified above. I'm assuming as a result, we don't need to mark as required here.

File: internal/cli/atlas/security/ldap/save.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/security/ldap/save.go:93]: Yes it does two times now 😄 , I am going to change this xD.

```
./bin/mongocli atlas security ldap save --help
Save an LDAP configuration.

Usage:
  mongocli atlas security ldap save [flags]

Flags:
      --authenticationEnabled       Specifies whether user authentication with LDAP is enabled.
      --authorizationEnabled        Specifies whether user authorization with LDAP is enabled.
      --authzQueryTemplate string   An LDAP query template that Atlas executes to obtain the LDAP groups to which the authenticated user belongs.
      --binPassword string          The password used to authenticate the bindUsername.
      --bindUsername string         The user DN that Atlas uses to connect to the LDAP server.
      --caCertificate string        CA certificate used to verify the identify of the LDAP server.
  -h, --help                        help for save
      --hostname string             The hostname or IP address of the LDAP server.
  -o, --output string               Output format.
                                    Valid values: json
      --port int                    The port to which the LDAP server listens for client connections. Default: 636 (default 636)
      --projectId string            Project ID to use. Overrides configuration file or environment variable settings.
```

**Review by colm-quinn — DISMISSED** [review]
LGTM, assuming the answer is yes to the questions above :)

File: internal/cli/atlas/security/ldap/save.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/security/ldap/save.go:106]: yes, when we set the default value we don't need to make the flag required

File: internal/cli/atlas/security/ldap/save.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/security/ldap/save.go:93]: new one

```
./bin/mongocli atlas security ldap save --help
Save an LDAP configuration.

Usage:
  mongocli atlas security ldap save [flags]

Flags:
      --authenticationEnabled       Specifies whether user authentication with LDAP is enabled.
      --authorizationEnabled        Specifies whether user authorization with LDAP is enabled.
      --authzQueryTemplate string   An LDAP query template that Atlas executes to obtain the LDAP groups to which the authenticated user belongs.
      --binPassword string          The password used to authenticate the bindUsername.
      --bindUsername string         The user DN that Atlas uses to connect to the LDAP server.
      --caCertificate string        CA certificate used to verify the identify of the LDAP server.
  -h, --help                        help for save
      --hostname string             The hostname or IP address of the LDAP server.
  -o, --output string               Output format.
                                    Valid values: json
      --port int                    The port to which the LDAP server listens for client connections. (default 636)
      --projectId string            Project ID to use. Overrides configuration file or environment variable settings.

Global Flags:
  -P, --profile string   Profile to use from your configuration file.
```

**Comment by andreaangiolillo** [comment]
@gssbzn tagging you here if you want to review this once you're back

File: internal/cli/atlas/security/ldap/save.go (hunk 1 above)
Comment (colm-quinn) [reply internal/cli/atlas/security/ldap/save.go:93]: Hehehe, nice inadvertent catch! :D

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for the updates

**Pull request merged**